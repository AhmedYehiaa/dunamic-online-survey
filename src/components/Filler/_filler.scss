
.filler {
  background: $main-color;
  height: 100%;
  border-radius: 0.6rem 0 0 0.6rem;
  transition: width .2s ease-in;
}

.filler__percentage {
  margin-left: 20px; 
}

