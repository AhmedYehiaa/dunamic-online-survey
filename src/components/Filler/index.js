import React from 'react';
import PropTypes from 'prop-types';

const Filler = (props) => {
  const { percentage } = props;
  return (
    <div className="filler" style={{ width: `${percentage}%` }}>
      <span className={`filler__percentage ${percentage > 0 ? 'text-white' : ''}`}>
        {percentage}
        %
      </span>
    </div>
  );
};

Filler.propTypes = {
  percentage: PropTypes.number.isRequired,
};
export default Filler;
