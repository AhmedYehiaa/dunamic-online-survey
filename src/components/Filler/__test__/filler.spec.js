import React from 'react';
import { shallow } from 'enzyme';
import Filler from '../index';

it('it should renders the filler component', () => {
  const wrapper = shallow(
    <Filler
      percentage={100}
    />,
  );
  expect(wrapper.children.length).toEqual(1);
  expect(wrapper).toMatchSnapshot();
});
