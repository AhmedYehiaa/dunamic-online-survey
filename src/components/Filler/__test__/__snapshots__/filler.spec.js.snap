// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should renders the filler component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Filler
    percentage={100}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <span
        className="filler__percentage text-white"
      >
        100
        %
      </span>,
      "className": "filler",
      "style": Object {
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          100,
          "%",
        ],
        "className": "filler__percentage text-white",
      },
      "ref": null,
      "rendered": Array [
        100,
        "%",
      ],
      "type": "span",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span
          className="filler__percentage text-white"
        >
          100
          %
        </span>,
        "className": "filler",
        "style": Object {
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            100,
            "%",
          ],
          "className": "filler__percentage text-white",
        },
        "ref": null,
        "rendered": Array [
          100,
          "%",
        ],
        "type": "span",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
